airbyte-ci/connectors/base_images/base_images/python/bases.py

```diff
@@ -13,8 +12,14 @@
 from base_images.root_images import PYTHON_3_10_14
 
 
-class AirbytePythonConnectorBaseImage(bases.AirbyteConnectorBaseImage):
+class AirbyteManifestOnlyConnectorBaseImage(bases.AirbyteConnectorBaseImage):
+    """ManifestOnly base image class, only used to fetch the registry."""
+
+    root_image: Final[published_image.PublishedImage] = PYTHON_3_10_14
```

Comment:
As `airbyte/source-declarative-manifest` is not built in this package (while other python base image are) I think the `root_image` attribute has no use in your case.
You could change `bases.AirbyteConnectorBaseImage` to not require this attribute and only set it on `AirbytePythonConnectorBaseImage`